addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1504,29 +1517,16 @@ export class OdooEditor extends EventTarget {
             n.textContent = '';
         });
         // Restore table contents removed by extractContents.
-        const tds = [...contents.querySelectorAll('td')].filter(n => !closestElement(n, 'table'));
-        let currentFragmentTr, currentTr;
-        const currentTd = closestElement(range.endContainer, 'td');
-        tds.forEach((td, i) => {
-            const parentFragmentTr = closestElement(td, 'tr');
-            // Skip the first and the last partially selected TD.
-            if (i && !(splitEndTd && i === tds.length - 1)) {
-                if (parentFragmentTr && parentFragmentTr !== currentFragmentTr && currentTr && [...parentFragmentTr.querySelectorAll('td')].every(td => tds.includes(td))) {
-                    currentTr.after(parentFragmentTr);
-                    currentTr = parentFragmentTr;
-                    parentFragmentTr.textContent = '';
-                } else {
-                    if (parentFragmentTr !== currentFragmentTr) {
-                        currentTr = currentTr
-                            ? currentTr.nextElementSibling
-                            : closestElement(range.endContainer, 'tr').nextElementSibling;
-                    }
-                    currentTr ? currentTr.prepend(td) : currentTd.after(td);
-                    td.textContent = '';
-                }
-            }
-            currentFragmentTr = parentFragmentTr;
-        });
+        const tds = [...contents.querySelectorAll('td')].filter(n => (
+            (startTrCells && startTrCells.includes(n) && !removeStartTable) ||
+            (endTrCells && endTrCells.includes(n) && !removeEndTable)
+        ));
+        tds.forEach(td => {
+            (startTrCells && startTrCells.includes(td)) ? startTd.after(td) : endTd.before(td);
+            td.textContent = '';
+            fillEmpty(td);
+            startTd = td;
```

Comment:
Its better to write the doc-string that justify the purpose & behavior you are trying to implement. Thanks!